src/polyswarm_api/api.py

```diff
@@ -489,8 +489,15 @@ def download(self, out_dir, hash_, hash_type=None):
         hash_ = resources.Hash.from_hashable(hash_, hash_type=hash_type)
         path = os.path.join(out_dir, hash_.hash)
         artifact = resources.LocalArtifact.from_path(self, path, create=True)
-        self.generator.download(hash_.hash, hash_.hash_type, handle=artifact).execute()
-        artifact.handle.close()
+
+        try:
+            self.generator.download(hash_.hash, hash_.hash_type, handle=artifact).execute()
+        except Exception as e:
+            os.remove(path)
```

Comment:
I think it might fail to remove the file before closing the handle. This might be an alternative:
```python
        try:
            self.generator.download(hash_.hash, hash_.hash_type, handle=artifact).execute()
        except Exception as e:
            artifact.handle.close()
            os.remove(path)
            raise e
        artifact.handle.close()
```

Reply:
Sure, forget that there's some OSs you can't remove files with open handles. Why can't everyone just use Linux....

Reply:
what about the other one in this diff